src/ProtoStruct.jl

```diff
@@ -181,8 +185,12 @@ macro proto(expr)
                         error("The supertype of a proto struct is not redefinable. Please restart your julia session.")
                     end
                     the_methods = collect(methods($name))
-                    Base.delete_method(the_methods[1])
-                    Base.delete_method(the_methods[2])
+                    if length(the_methods) >= 1
+                        Base.delete_method(the_methods[1])
+                    end
+                    if length(the_methods) >= 2
+                        Base.delete_method(the_methods[2])
+                    end
```

Comment:
If we deleted a method in the if clause above this might not delete the second method if its the only remaining method, if I got this right

Reply:
At first I thought the same when I saw these changes, but then I realized that `the_methods`  is not modified by the deletion